recipes/arg_router/all/conanfile.py

```diff
@@ -1,57 +1,88 @@
 import os
 
 from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
 from conan.tools.build import check_min_cppstd
-from conan.tools.files import get, rmdir
-from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir
+from conan.tools.scm import Version
 
+required_conan_version = ">=1.53.0"
 
-class arg_routerRecipe(ConanFile):
+
+class ArgRouterConan(ConanFile):
     name = "arg_router"
     license = "BSL-1.0"
     homepage = "https://github.com/cmannett85/arg_router"
     url = "https://github.com/conan-io/conan-center-index"
     description = "C++ command line argument parsing and routing."
     topics = ("cpp", "command-line", "argument-parser", "header-only")
-
-    # CMake >= 3.18 is required https://github.com/cmannett85/arg_router/blob/449567723d6c0e9db0a4c89277066c9a53b299fa/CMakeLists.txt#L5
-    tool_requires = "cmake/3.25.3"
-    requires = "boost/1.81.0", "span-lite/0.10.3"
-
     settings = "os", "arch", "compiler", "build_type"
     package_type = "header-library"
-    generators = "CMakeDeps", "CMakeToolchain"
-    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _min_compilers_version(self):
+        return {
+            "apple-clang": "10",
+            "clang": "5",
+            "gcc": "10",
+            "msvc": "191",
+            "Visual Studio": "15",
```

Comment:
I've only ever tested it on MSVC 19.3x and VS 17.x - I've found MSVC to have poor TMP support.